src/zenhub.py

```diff
@@ -67,21 +68,106 @@ def _generate_url(self):
         _url = self.access_params['options']['server']
         return os.path.join(_url, self.repo_id, 'issues', self.issue)
 
+    def _get_pipeline_ids(self):
+        # Determine the valid pipeline IDs for this repo.
+        logger.info(f'Retrieving pipeline ids for {self.repo_name}.')
+        response = requests.get(f'{self.url}{self.repo_id}/board', headers=self.headers)
+
+        if response.status_code == 200:
+            logger.info(f'Successfully retrieved pipeline ids for {self.repo_name}.')
+            data = response.json()
+            ids = {pipeline['name']: pipeline['id'] for pipeline in data['pipelines']}
+            return ids
+        else:
+            logger.info(
+                f'Error in retreiving pipeline ids. Status Code: {response.status_code}. Reason: {response.reason}')
+            raise RuntimeError(
+                f'Error in retreiving pipeline ids. Status Code: {response.status_code}. Reason: {response.reason}')
+
+
+class ZenHubIssue(Issue):
+
+    def __init__(self, key: str = None, repo: str = None, org: str = None, response: dict = None):
+        """
+        Create an Issue object from an issue key and repo name or from a portion of a ZenHub API response
+
+        :param key: If this and repo_name are specified, make an API call searching by this issue key
+        :param repo: If this and key are specified, make an API call searching in this repo
+        :param org: The organization to which the repo belongs, e.g. ucsc-cgp
+        :param response: If specified, don't make a new API call but use this response from an earlier one
+        """
+
+        super().__init__()  # Initiate the parent class, Issue
```

Comment:
Wouldn't write a comment here. You could do `Issue.__init__()` instead.

Reply:
removed comment